Ubuntu: Support debian packaging for Ubuntu-based CI server

leemgs opened the pull request:
Fixes issue #233, #235.

This commit is to support a debian package management for Ubuntu server.
This files is installed in /var/www/html folder by default.

Now, It is a time to provide a debian packging for server administrator that
want to install more easily the TAOS-CI system on a Ubuntu distribution.

**Changes proposed in this PR:**

* Version2:
1. Added CMakeLists.txt to support cmake-based debian packaging
2. Fixed typo in CODEOWNERS file.
3. Updated debian/rules file for cmake build

* Version1:
1. Added required files in debian/ subdirectory for debian packaging
2. Added ./debian/folder in ./.github/CODEOWNERS for auto reviewer list

**How to evaluate:**
$ debuild -uc -us
$ ls -al ../taos-ci*.deb

Signed-off-by: Geunsik Lim <leemgs@gmail.com>

myungjoo commented on .github/CODEOWNERS:
```diff
@@ -6,11 +6,12 @@
 
 # For /packaging/, reviewers are limited to those who have some RPM packaging experiences
 /packaging/      @myungjoo @leemgs @again4you @wook16-song
+/debian/         @myungjoo @leemgs @again4you @wook16-song
```
Who is wook16-song? Do you mean @wooksong ? :)

myungjoo commented on debian/rules:
```diff
@@ -0,0 +1,17 @@
+	mkdir -p    debian/taos-ci/var/www/html/TAOS-CI
+	cp -arf ./ci debian/taos-ci/var/www/html/TAOS-CI/
+	dh_gencontrol
+	dh_builddeb
```
uh...

Doesn't
```
%:
        dh $@
```
fit for your needs here? (You usually don't need clean, build section specified)

It'd be better to have a common install script for both Tizen and Ubuntu (e.g., CMaktLists.txt with INSTALL sections). Having too specific script in debian/rules isn't too good for porting.

leemgs replied:
We do not need a build section becaue TAOS-CI is written by Script languages (e.g., PHP, Bash, and Python). So, this PR is not used CMakeList.txt by default.

leemgs commented:
* After this PR version 2:
```bash
$ tree
.
├── taos-ci_1.0-1_all.deb
└── tmp
    ├── DEBIAN
    │   ├── control
    │   └── md5sums
    ├── usr
    │   └── share
    │       └── doc
    │           └── taos-ci
    │               ├── changelog.Debian.gz
    │               └── copyright
    └── var
        └── www
            └── html
                └── TAOS-CI
                    └── ci
                        ├── Doxyfile
                        ├── Doxyfile.ci
                        ├── Doxyfile.prj
                        ├── api_collection.sh
                        ├── body.php
                        ├── book-hard-copy-cover1.odt
                        ├── book-hard-copy-date.txt
                        ├── book-hard-copy-prj-generate.sh
                        ├── checker-issue-comment.sh
                        ├── checker-pr-audit-async.sh
                        ├── checker-pr-audit.sh
                        ├── checker-pr-comment.sh
                        ├── checker-pr-format-async.sh
                        ├── checker-pr-format.sh
                        ├── checker-pr-signed-off-by.sh
                        ├── ci01.png
                        ├── ci02.png
                        ├── ci03.png
                        ├── coding-convention.md
                        ├── config-environment.sh
                        ├── config-plugins-audit.sh
                        ├── config-plugins-format.sh
                        ├── config-server-administrator.sh
                        ├── config-webhook.json
                        ├── contributing.md
                        ├── coverage-generator.sh
                        ├── doxygen-documentation.md
                        ├── format-checker.sh
                        ├── gen-diff-patch.sh
                        ├── gen-md-toc.sh
                        ├── home-16.gif
                        ├── home-24.gif
                        ├── how-to-archive-large-data.md
                        ├── how-to-build-gbs-pdebuild.md
                        ├── how-to-good-commit-message.md
                        ├── how-to-install-taos-ci.md
                        ├── how-to-setup-taos-ci-server.md
                        ├── how-to-unit-testing.md
                        ├── how-to-use-lcov-to-generate-code-coverage-stat.md
                        ├── how-to-use-taos-ci-module.md
                        ├── index.html
                        ├── install-packages.sh
                        ├── maintenance.md
                        ├── menu.html
                        ├── monitor-icon.png
                        ├── monitor.php
                        ├── newline-checker.sh
                        ├── nobody-checker.sh
                        ├── pr-audit-build-tizen.sh
                        ├── pr-audit-build-ubuntu.sh
                        ├── pr-audit-build-yocto.sh
                        ├── pr-audit-resource.sh
                        ├── pr-comment.sh
                        ├── pr-format-clang.sh
                        ├── pr-format-cppcheck.sh
                        ├── pr-format-doxygen-build.sh
                        ├── pr-format-doxygen-tag.sh
                        ├── pr-format-exclusive-vio.sh
                        ├── pr-format-executable.sh
                        ├── pr-format-file-size.sh
                        ├── pr-format-hardcoded-path.sh
                        ├── pr-format-indent.sh
                        ├── pr-format-misspelling.sh
                        ├── pr-format-newline.sh
                        ├── pr-format-nobody.sh
                        ├── pr-format-pylint.sh
                        ├── pr-format-rpm-spec.sh
                        ├── pr-format-timestamp.sh
                        ├── pr-report.sh
                        ├── self-assessment-before-submitting-pr.md
                        ├── signed-off-by-checker.sh
                        ├── sleep.png
                        ├── sorry.gif
                        ├── taos-platform-dmesg-error.sh
                        ├── taos-platform-dmesg-oom.sh
                        ├── taos-platform-free-space.sh
                        ├── taos-platform-obs-build-error.sh
                        ├── taos-platform-swap-status.sh
                        ├── test-coverage-gcov.sh
                        ├── test.php
                        ├── top.php
                        ├── unittestcoverage.py
                        ├── webhook-flow.png
                        └── webhook.php

11 directories, 89 files
```

leemgs commented:
**Changes proposed in this PR:**

* Version2:
1. Added CMakeLists.txt to support cmake-based debian packaging
2. Fixed typo in CODEOWNERS file.
3. Updated debian/rules file for cmake build

* Version1:
1. Added required files in debian/ subdirectory for debian packaging
2. Added ./debian/folder in ./.github/CODEOWNERS for auto reviewer list

**How to evaluate:**
$ debuild -uc -us
$ ls -al ../taos-ci*.deb

leemgs commented:
Comments are applied. The aging test is successfully done.
